pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -2658,9 +2726,15 @@ protected void handleAddSubscriptionToTxn(CommandAddSubscriptionToTxn command) {
         TransactionMetadataStoreService transactionMetadataStoreService =
                 service.pulsar().getTransactionMetadataStoreService();
 
-        transactionMetadataStoreService.addAckedPartitionToTxn(txnID,
-                MLTransactionMetadataStore.subscriptionToTxnSubscription(command.getSubscriptionsList()))
-                .whenComplete(((v, ex) -> {
+        verifyTxnOwnership(txnID)
```

Comment:
Is it better to move the txn owner check to the transaction metadata store service?

Reply:
I moved only the part related to the interaction with txnMeta inside the txn metadata store service